pipeline/hudi-connector/src/main/scala/org/sunbird/obsrv/streaming/HudiConnectorStreamTask.scala

```diff
@@ -0,0 +1,153 @@
+    val rowType = schemaParser.rowTypeMap(dataset)
+    val avroSchema = AvroSchemaConverter.convertToSchema(rowType, dataset.replace("-", "_"))
+    conf.setString(FlinkOptions.PATH.key, s"${config.hudiBasePath}/${datasetSchema.schema.table}")
+    conf.setString(FlinkOptions.TABLE_NAME, datasetSchema.schema.table)
+    conf.setString(FlinkOptions.RECORD_KEY_FIELD.key, datasetSchema.schema.primaryKey)
+    conf.setString(FlinkOptions.PRECOMBINE_FIELD.key, datasetSchema.schema.timestampColumn)
+    conf.setString(FlinkOptions.PARTITION_PATH_FIELD.key, datasetSchema.schema.partitionColumn)
+    conf.setString(FlinkOptions.SOURCE_AVRO_SCHEMA.key, avroSchema.toString)
+
+    val partitionField = datasetSchema.schema.columnSpec.filter(f => f.name.equalsIgnoreCase(datasetSchema.schema.partitionColumn)).head
+    if(partitionField.`type`.equalsIgnoreCase("timestamp") || partitionField.`type`.equalsIgnoreCase("epoch")) {
+      conf.setString(FlinkOptions.PARTITION_PATH_FIELD.key, datasetSchema.schema.partitionColumn + "_partition")
+    }
+
+    if (config.hmsEnabled) {
+      conf.setString("hive_sync.table", datasetSchema.schema.table)
+    }
+  }
+
+  private def setHudiBaseConfigurations(conf: Configuration): Unit = {
+    conf.setString(FlinkOptions.TABLE_TYPE.key, config.hudiTableType)
+    conf.setBoolean(FlinkOptions.METADATA_ENABLED.key, true)
+    conf.setDouble(FlinkOptions.WRITE_BATCH_SIZE.key, 0.1)
+    conf.setBoolean(FlinkOptions.COMPACTION_SCHEDULE_ENABLED.key, config.hudiCompactionEnabled)
+    conf.setInteger("write.tasks", config.hudiWriteTasks)
+    conf.setInteger(FlinkOptions.COMPACTION_DELTA_COMMITS, 2)
+    conf.setString(FlinkOptions.COMPACTION_TRIGGER_STRATEGY, "num_or_time")
+    conf.setBoolean(FlinkOptions.COMPACTION_ASYNC_ENABLED, true)
+    conf.setInteger(FlinkOptions.BUCKET_ASSIGN_TASKS, 1)
+    conf.setInteger(FlinkOptions.COMPACTION_TASKS, 1)
+    conf.setString("hoodie.fs.atomic_creation.support", "s3a")
+    conf.setString(FlinkOptions.HIVE_SYNC_TABLE_PROPERTIES, "hoodie.datasource.write.drop.partition.columns=true")
+    conf.setBoolean(DROP_PARTITION_COLUMNS.key, true)
+    conf.setBoolean(SCHEMA_ALLOW_AUTO_EVOLUTION_COLUMN_DROP.key(), true); // Enable dropping columns
+    conf.setBoolean(SCHEMA_EVOLUTION_ENABLE.key(), true); // Enable schema evolution
+    conf.setString(FlinkOptions.PAYLOAD_CLASS_NAME, "org.apache.hudi.common.model.PartialUpdateAvroPayload")
+
+    if (config.hmsEnabled) {
+      conf.setBoolean("hive_sync.enabled", config.hmsEnabled)
+      conf.setString(FlinkOptions.HIVE_SYNC_DB.key(), config.hmsDatabaseName)
+      conf.setString("hive_sync.username", config.hmsUsername)
+      conf.setString("hive_sync.password", config.hmsPassword)
+      conf.setString("hive_sync.mode", "hms")
+      conf.setBoolean("hive_sync.use_jdbc", false)
+      conf.setString(FlinkOptions.HIVE_SYNC_METASTORE_URIS.key(), config.hmsURI)
+      conf.setString("hoodie.fs.atomic_creation.support", "s3a")
```

Comment:
lines: 111, 126: need to load from config?

Reply:
@ravismula These are more or less mandatory configurations. The atomic creation support may be is applicable only on S3.